internal/logging/logr.go

```diff
@@ -59,3 +53,53 @@ func FromContext(ctx context.Context) logr.Logger {
 
 	return log
 }
+
+// sink implements logr.LogSink using two function pointers.
+type sink struct {
+	depth     int
+	verbosity int
+	names     []string
+	values    []interface{}
+
+	// TODO(cbandy): add names or frame to the functions below.
+
+	fnError func(error, string, ...interface{})
+	fnInfo  func(int, string, ...interface{})
+}
+
+var _ logr.LogSink = (*sink)(nil)
```

Comment:
How did we decide to use Zap elsewhere?

Reply:
(It is interesting that we have this sort of set up as if we were going to have log levels, but essentially we just have a bool for debug on/off in main.go.)

(Also, this logging is 100% from your branch, so I'm glad it looks good :) )

Reply:
> How did we decide to use Zap elsewhere?

I think we used it because controller-runtime provides some [glue/enhancements](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/log/zap/zap.go) for it, and/or it's part of the Kubebuilder [project template](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/main.go#L32).